### PR TITLE
Add link to android stylesheet to fix Andika (BL-4947)

### PR DESCRIPTION
**Pull request by andrew-polk**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/1831)

### CODE REVIEW

File: src/BloomExe/Book/BookCompressor.cs [hunk 1]
```diff
@@ -264,13 +264,20 @@ private static string GetMetaJsonModfiedForTemplate(string path)
...
 			var link = dom.CreateElement("link");
-			XmlUtils.GetOrCreateElement(dom, "html", "head").AppendChild(link);
 			link.SetAttribute("rel", "stylesheet");
 			link.SetAttribute("href", "readerStyles.css");
 			link.SetAttribute("type", "text/css");
+			head.AppendChild(link);
+
+			link = dom.CreateElement("link");
+			link.SetAttribute("rel", "stylesheet");
+			link.SetAttribute("href", "file:///android_asset/book support files/android.css");
```
Comment (hatton) [inline src/BloomExe/Book/BookCompressor.cs]: It seems like we're being misleadingly specific by calling out "android". ".bloomd" isn't meant to be ".blooma".
Similarly, it seems awkward for Bloom Desktop to know the path inside of BloomReader in which to get this file.

**Comment by andrew-polk** [comment]
*[src/BloomExe/Book/BookCompressor.cs, line 278 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831#-Ksu1-KeUigMXQPDrtBe-r1-278:-Ksu6fD0ULGzem83FFFA:bxandv1) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c370a8dc8d9d3f0535d697eb4ddbe6e53dae6c19/src/BloomExe/Book/BookCompressor.cs#L278)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

It seems like we're being misleadingly specific by calling out "android". ".bloomd" isn't meant to be ".blooma".
Similarly, it seems awkward for Bloom Desktop to know the path inside of BloomReader in which to get this file.
</blockquote></details>

android.css exists already. But I can certainly rename it.

Alternative approaches:
- put css from android.css into readerstyles. Con: the css has android-context file paths
- add the stylesheet link in BR. Con: it would be the only thing parsing the html (unless we did it by string manipulation)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831)*

**Comment by JohnThomson** [comment]
I don't fully understand this. Why do we need android.css (nested several folders deep) when we already created readerStyles.css on purpose to contain any special styles needed by reader programs? At least as long as we only have one reader program it seems one special stylesheet for it should be enough.

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/Book/BookCompressor.cs, line 278 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831#-Ksu1-KeUigMXQPDrtBe-r1-278:-Ksu7-tCkryyb8f6jhNK:bqhgxdt) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c370a8dc8d9d3f0535d697eb4ddbe6e53dae6c19/src/BloomExe/Book/BookCompressor.cs#L278)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

android.css exists already. But I can certainly rename it.

Alternative approaches:
- put css from android.css into readerstyles. Con: the css has android-context file paths
- add the stylesheet link in BR. Con: it would be the only thing parsing the html (unless we did it by string manipulation)
</blockquote></details>

How does this even work, anyway? I don't see any new code to create or copy this stylesheet into the book...nor do I see any such folder or file in existing books...or is this supposed to be an absolute path that will somehow find a file that is part of BR?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831#-:-Ksu6SFrvqv-NxhcA76z:b-rfs4zy)*

**Comment by JohnThomson** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/Book/BookCompressor.cs, line 278 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831#-Ksu1-KeUigMXQPDrtBe-r1-278:-Ksu7uPC-XhWxtSKCT2o:b6vvzwi) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c370a8dc8d9d3f0535d697eb4ddbe6e53dae6c19/src/BloomExe/Book/BookCompressor.cs#L278)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

How does this even work, anyway? I don't see any new code to create or copy this stylesheet into the book...nor do I see any such folder or file in existing books...or is this supposed to be an absolute path that will somehow find a file that is part of BR?
</blockquote></details>

Ah, yes, I see, it's a file that ships with BR. So we're depending on an assumption about the browser starting to search for file: urls at the root install directory of the app? Feels a lot more robust to me for the reader to insert a link using its definite knowledge of the file's absolute path.
It's a pretty easy string manipulation, I would think...e.g., find </head and add the link before it. There's already code that assembles the HTML for each page as a string, so one more tweak to it wouldn't be too painful. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831)*

**Comment by JohnThomson** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831)*

**Comment by andrew-polk** [comment]
*[src/BloomExe/Book/BookCompressor.cs, line 278 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831#-Ksu1-KeUigMXQPDrtBe-r1-278:-Ksu8lMjfZIMfF61kg85:b-mlmj2u) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/c370a8dc8d9d3f0535d697eb4ddbe6e53dae6c19/src/BloomExe/Book/BookCompressor.cs#L278)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

Ah, yes, I see, it's a file that ships with BR. So we're depending on an assumption about the browser starting to search for file: urls at the root install directory of the app? Feels a lot more robust to me for the reader to insert a link using its definite knowledge of the file's absolute path.
It's a pretty easy string manipulation, I would think...e.g., find </head and add the link before it. There's already code that assembles the HTML for each page as a string, so one more tweak to it wouldn't be too painful. 
</blockquote></details>

Ok, I can do the string manipulation to add it on the BR side.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1831)*

**Comment by andrew-polk** [comment]
Closing in favor of handling this on the BR side.

**Pull request closed without merging**